docs/cli-compile.md

```diff
@@ -11,3 +11,22 @@ It provides the following options:
 The `-n` or `--name` option provides control over the name of the top level
 kernel function generated to wrap the `decl`s. The default kernel name is
 `kernel`.
+
+## Output file
+
+The `-o` or `--out` option tells the compiler to place the output in a new
+file `<out>` instead of printing to STDOUT.
+
+## Backend
+
+The `-b` or `--backend` allows changing the backend used for generating code.
+Support backends are:
+
+1. `vivado`: Generates code compatible with Vivado HLS.
+2. `c++`: Generate code that can be compiled into an executable. See [this](cpp-runnable.md).
+
+## `run`
+
+The subcommand `run` requires the `-o` to be present. It automatically generates
+a CPP file and links it with the json data parser. See [this](cpp-runnable.md)
+for more information.
```

Comment:
This is just an aesthetic UI argument, but it's weird that the CLI has an *optional* subcommand. That is, most CLIs I know either have subcommands or don't—`git` has them, obviously, and there's no way to *just* invoke `git` without a subcommand. Same with `svn` before it.

For whatever reason, this feels to me like it would be just fine as a flag. That is, `fuse --run source.file` seems to work just fine and avoids the "optional subcommand" thing.

Reply:
So “run” implies a bunch of other things such as backend=c++ and the fact that -o is required.

I tried making —run a flag but the code for enforcing the above constraints is non local and annoying to read. For example, the code has to say if mode=run then validate that -o is not None and backend=c++. Note that having a subcommand doesn’t stop us from specific common things like kernelName

For uniformity’s sake, I can make a fuse compile command. Would that work?